rocAL/include/pipeline/tensor.h

```diff
@@ -191,7 +191,7 @@ class TensorInfo {
     }
     void set_tensor_layout(RocalTensorlayout layout) {
         if (layout == RocalTensorlayout::NONE) return;
-        if (_layout != layout && _layout != RocalTensorlayout::NONE) {  // If layout input and current layout's are different modify dims accordingly
+        if (_layout != layout && _layout != RocalTensorlayout::NONE && (_num_of_dims > 3)) {  // If layout input and current layout's are different modify dims accordingly
```

Comment:
Is there a case for num_dims < 3 and how is it handled

Reply:
We only support layout change for tensors with 3 dims and above. For 2 dims, there is no layout switch supported